framework/mesh/mesh_generator/mesh_generator.cc

```diff
@@ -198,9 +198,54 @@ MeshGenerator::PartitionMesh(const UnpartitionedMesh& input_umesh, int num_parti
   std::vector<int64_t> cell_pids =
     partitioner_->Partition(cell_graph, cell_centroids, num_partitions);
 
+  RebalancePartitions(cell_pids, num_partitions);
+
   return cell_pids;
 }
 
+void
+MeshGenerator::RebalancePartitions(std::vector<int64_t>& cell_pids, int num_partitions)
+{
+  // Count the number of cells in each partition
+  std::vector<int> cells_per_partition(num_partitions, 0);
+  for (int partition : cell_pids)
+    cells_per_partition[partition]++;
+
+  // Check if any partition has zero cells
+  if (not std::any_of(cells_per_partition.begin(),
+                      cells_per_partition.end(),
+                      [](int count) { return count == 0; }))
+  {
+    return;
+  }
```

Comment:
I'm not sure if it is a coding standard thing, but should we use braces for one-liners like this?

Reply:
If the `if` statement has been a single line, I would agree that it wouldn't be necessary to use `{ }`. Since the `if` statement is multi-line, I think the braces help with clarity.

Reply:
I agree with Daryl. The if statement that s long, so the curly braces help with readability in my opinion

Reply:
I understood that the discussion was about:

```
if (not std:any())
  return;
```
vs.
```
if (not std::any_of())
{
  return;
}
```

Are you guys arguing that the length of the conditional of the if-statement would determine the use of the braces for the "body"?

Reply:
I believe the discussion is about curly braces or not, for one-liners. I suppose we are pondering when to allow the use of `{...}` when the conditional is complex and we think the use of braces enhance readability. To me, they do.

on readability, is this below not equivalent and a bit shorter?
```
if (std::none_of(cells_per_partition.begin(),
                 cells_per_partition.end(),
                 [](int count) { return count == 0; }))
```

Reply:
> Are you guys arguing that the length of the conditional of the if-statement would determine the use of the braces for the "body"?

If the body of the conditional spans multiple lines, I always use braces. If the body fits on a single line, I don't use braces. For me, it helps with clarity when I'm quickly scanning code.

Reply:
> on readability, is this below not equivalent and a bit shorter?
> 
> ```
> if (std::none_of(cells_per_partition.begin(),
>                  cells_per_partition.end(),
>                  [](int count) { return count == 0; }))
> ```

It is. I'll make this change.

Reply:
Fixed.